tapir/coop/services/ResignMemberService.py

```diff
@@ -0,0 +1,84 @@
+import datetime
+
+from tapir.accounts.models import TapirUser
+from tapir.coop.models import ResignedMembership, ShareOwnership
+from tapir.coop.forms import MembershipCancelForm
+from tapir.shifts.models import (
+    ShiftAttendanceTemplate,
+    ShiftAttendance,
+    ShiftAccountEntry,
+)
+from tapir.shifts.services.frozen_status_service import FrozenStatusService
+from tapir.shifts.models import ShiftUserData, ShiftAttendanceMode
+
+from tapir.utils.shortcuts import get_timezone_aware_datetime
+
+
+class ResignMemberService:
+    @staticmethod
+    def update_shifts_and_shares(
+        form: MembershipCancelForm, member: ResignedMembership
+    ):
+        tapir_user: TapirUser = getattr(member.share_owner, "user", None)
+        if not tapir_user:
+            print("Couldn't find an existing Tapir user.")
+            return
+        shares = ShareOwnership.objects.filter(share_owner=member.share_owner)
+        if (
+            member.willing_to_gift_shares_to_coop
+            or member.transfering_shares_to is not None
```

Comment:
I would wrap this in a function like `is_resignation_happening_right_away` to make it clear what we are checking here